michelson.md

```diff
@@ -2362,6 +2361,26 @@ It has an untyped and typed variant.
   rule <k> #MakeFresh(or _:AnnotationList T1 T2) => Right #MakeFresh(T2) ... </k>
 ```
 
+We implement fresh lambdas as fresh uninterpreted functions.
+
+```symbolic
+  rule <k> #MakeFresh(lambda    _:AnnotationList T1 T2)
+        => #Lambda(#Name(T1), #Name(T2), { #Uninterpreted(!Id, #Name(T1), #Name(T2)) })
+           ...
+       </k>
+
+  syntax Instruction ::= #Uninterpreted(id: Int, arg: TypeName, return: TypeName)
+  syntax Data ::= uninterpreted(id: Int, arg: Data) [function, functional, no-evaluators]
+  // ------------------------------------------------------------------------------------
+  rule <k> #Uninterpreted(Id, ArgT, RetT)
```

Comment:
Can we do the `#Bottom` thing here too? Might be good to use one method consistently.